app/domain/authentication/authn_jwt/signing_key/create_jwks_from_http_response.rb

```diff
@@ -0,0 +1,60 @@
+module Authentication
+  module AuthnJwt
+    module SigningKey
+      # CreateJwksFromHttpResponse command class is responsible to create jwks object from http response
+      CreateJwksFromHttpResponse ||= CommandClass.new(
+        dependencies: {
+          logger: Rails.logger,
+          jwk_set_class: JSON::JWK::Set
+        },
+        inputs: %i[http_response]
+      ) do
+        def call
+          validate_response_exists
+          validate_response_has_a_body
+          create_jwks_from_http_respnse
+        end
+
+        private
+
+        def validate_response_exists
+          raise Errors::Authentication::AuthnJwt::MissingHttpResponse if @http_response.blank?
+        end
+
+        def validate_response_has_a_body
+          raise Errors::Authentication::AuthnJwt::InvalidHttpResponseFormat unless @http_response.respond_to?(:body)
+        end
+
+        def create_jwks_from_http_respnse
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CreatingJwksFromHttpResponse.new)
+
+          response_body = @http_response.body
```

Comment:
i would move this to a memoizing method:
```
def response_body
  @response_body ||= @http_response.body
end
```

Reply:
this way you don't need to pass these objects to `parse_jwks_response` and you can call them directly in the method implementation

Reply:
Done